app/src/main/java/com/pvp/app/ui/screen/friends/FriendsViewModel.kt

```diff
@@ -0,0 +1,142 @@
+package com.pvp.app.ui.screen.friends
+
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.pvp.app.api.FriendService
+import com.pvp.app.api.UserService
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+
+@HiltViewModel
+class FriendsViewModel @Inject constructor(
+    private val friendService: FriendService,
+    private val userService: UserService
+) : ViewModel() {
+    val toastMessage = mutableStateOf<String?>(null)
+    private val _isRequestSent = MutableStateFlow(false)
+    val isRequestSent = _isRequestSent.asStateFlow()
+
+    val user = userService.user.stateIn(
+        viewModelScope,
+        started = SharingStarted.WhileSubscribed(5000),
+        initialValue = null
+    )
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    val friendObject = user.flatMapLatest { user ->
+        user?.email?.let { userEmail ->
+            friendService.get(userEmail)
+        } ?: flowOf(null)
+    }.stateIn(
+        viewModelScope,
+        started = SharingStarted.WhileSubscribed(5000),
+        initialValue = null
+    )
+
+    init {
+        viewModelScope.launch {
+            user.collect { user ->
+                user?.email?.let { email ->
+                    friendService.createFriendObject(email)
+                }
+            }
+        }
+    }
+
+    fun createFriendObject(email: String) {
+        viewModelScope.launch {
+            friendService.createFriendObject(email)
+        }
+    }
+
+    fun addFriend(friendEmail: String){
+        viewModelScope.launch {
+            if (friendEmail.isEmpty()) {
+                toastMessage.value = "Please enter an email"
+                _isRequestSent.value = false
+                return@launch
+            }
+
+            val userEmail = user.value?.email ?: return@launch
+            val friendObject = friendService.get(userEmail).firstOrNull() ?: return@launch
```

Comment:
Variables should be split on chained method calls. After split, values will become multi-line and will have to be separated from above/below.